website_sale_secondary_unit/controllers/main.py

```diff
@@ -33,3 +33,10 @@ def cart_update_json(
             set_qty=set_qty,
             display=display,
         )
+
+    def _prepare_product_values(self, product, category, search, **kwargs):
+        res = super()._prepare_product_values(product, category, search, **kwargs)
+        res["secondary_uom_ids"] = product.secondary_uom_ids.filtered(
+            lambda su: su.active and su.is_published
```

Comment:
su.active is not needed

Reply:
If we don't set it, as you can see in the description, non published and archived secondary_units were visible at e-comerce

Reply:
I thought that ORM made it by default when you use filtered

Reply:
Yes but the field has this context https://github.com/OCA/product-attribute/blob/c570ac4984c83bcae877208a9ec1ebcbfd707f1a/product_secondary_unit/models/product_template.py#L14 to allow seeing the archived secondary units at product level.

Reply:
:+1: